src/app/analytics/analytics.factory.js

```diff
@@ -475,42 +477,55 @@ const analyticsFactory = /* @ngInject */ function ($window, $timeout, sessionSer
         // Error caught in analyticsFactory.setPurchaseNumber
       }
     },
-    transactionEvent: function (purchaseData, cartData) {
+    transactionEvent: function (purchaseData) {
       try {
-        const cartObject = cartData.items.map((cartItem) => {
-          return {
-            name: cartItem.displayName.toLowerCase(),
-            id: cartItem.code,
-            price: cartItem.amount.toString(),
-            branch: 'cru',
-            category: cartItem.designationType.toLowerCase(),
-            variant: cartItem.frequency.toLowerCase(),
-            quantity: '1'
-          }
-        })
-
-        if (typeof $window.dataLayer !== 'undefined') {
-          $window.dataLayer.push({
-            event: 'transaction',
-            paymentType: purchaseData.paymentMeans['account-type'].toLowerCase(),
-            ecommerce: {
-              currencyCode: 'USD',
-              purchase: {
-                actionField: {
-                  id: purchaseData.rawData['purchase-number'],
-                  affiliation: undefined,
-                  revenue: purchaseData.rawData['monetary-total'][0].amount.toString(),
-                  shipping: undefined,
-                  tax: undefined,
-                  coupon: undefined
-                },
-                products: [
-                  ...cartObject
-                ]
-              }
+        // Parse the cart object of the last purchase
+        const transactionCart = JSON.parse(localStorage.getItem('transactionCart'))
+        // The purchaseId number from the last purchase
+        const lastTransactionId = localStorage.getItem('transactionId')
+        // The purchaseId number from the pruchase data being passed in
+        const currentTransactionId = purchaseData.rawData['purchase-number']
+        // If the lastTransactionId and the current one do not match, we need to send an analytics event for the transaction
+        if (lastTransactionId !== currentTransactionId) {
+          // Set the transactionId in localStorage to be the one that is passed in
+          localStorage.setItem('transactionId', currentTransactionId)
```

Comment:
So, it looks like `transactionId` is never cleared from local storage. I wonder if this could introduce a security hole, especially on public computers. Maybe it should be cleared on sign-in and/or sign-out events?

Reply:
I also thought about doing sessionStorage instead, do you think that would be better?

Reply:
How would the thank you summary load in a new browser session?

Reply:
You can disregard that last part haha. Should I go with sessionStorage instead so we make sure it's cleared after their session ends regardless of being signed in or not?

Reply:
Actually, I think that is the way to go because we do the same thing for the ```lastPurchaseLink```, ```storedCvvs```, etc

Reply:
So, I thought of one way that might hit the thank you page without going through checkout, which is site crawlers. You can go to https://give-stage2.cru.org/thank-you.html directly without a cart. It throws an error, but we should probably also account for that case of bots/web crawlers.